# GRUPO 1 - WORKGROUP 4
*Integrantes: Keyth Hurtado, Fátima Trujillo y Melani Geng*

En este script procesamos la base de datos de la Comisión de la Verdad y Reconciliación (CVR). Esta base de datos es el registra actos de violencia perpetrados contra individuos.

In [3]:
#Primero importamos las librerías necesarias
import pandas as pd
import numpy as np
import os
import pyreadstat

In [4]:
#Luego cargamos la base de datos 
cvr, meta =  pyreadstat.read_sav( r"../../data/actos_est.sav")
cvr

IDEVENTO   IDACTO  IDACTOR IDTIPOAC  LDS_LDT  LDT  LRC  LSE  MAE  MAT  \
0      1000001  14867.0  54406.0      LDS      1.0  0.0  0.0  0.0  0.0  0.0   
1      1000001  14867.0  54407.0      LDS      1.0  0.0  0.0  0.0  0.0  0.0   
2      1000001  20535.0  74109.0      LDT      0.0  1.0  0.0  0.0  0.0  0.0   
3      1000001  20535.0  74110.0      LDT      0.0  1.0  0.0  0.0  0.0  0.0   
4      1000003  14834.0  54220.0      LDT      0.0  1.0  0.0  0.0  0.0  0.0   
...        ...      ...      ...      ...      ...  ...  ...  ...  ...  ...   
36014  1200682  13183.0  48743.0      MAE      0.0  0.0  0.0  0.0  1.0  0.0   
36015  1991433    630.0   2473.0      LDT      0.0  1.0  0.0  0.0  0.0  0.0   
36016  1991433    632.0   2478.0      TTR      0.0  0.0  0.0  0.0  0.0  0.0   
36017  1991433    640.0   2509.0      MAE      1.0  0.0  0.0  0.0  1.0  0.0   
36018  1991433    644.0   2528.0      LDT      0.0  1.0  0.0  0.0  0.0  0.0   

       ...  T08  T09  T10  T11  T12  T13  T14  T15  T16  T90  
0      ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
1      ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
2      ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
3      ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
4      ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
...    ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  
36014  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
36015  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
36016  ...  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  
36017  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
36018  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  

[36019 rows x 118 columns]

Se puede ver que tenemos 118 variables, con 36018 observaciones.

In [5]:
cvr.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36019 entries, 0 to 36018
Columns: 118 entries, IDEVENTO to T90
dtypes: datetime64[ns](3), float64(106), object(9)
memory usage: 32.4+ MB


## 1. Mostar las etiquetas de valor y etiquetas de las variables

In [6]:
# Extraemos las etiquetas de variable en var_labels
var_labels=dict(zip(meta.column_names, meta.column_labels))
var_labels

{'IDEVENTO': 'ID DEL EVENTO',
 'IDACTO': 'ID DEL ACTO',
 'IDACTOR': 'ID DEL ACTOR',
 'IDTIPOAC': 'TIPO DE ACTO',
 'LDS_LDT': 'Desaparición forzada',
 'LDT': 'Detenciones',
 'LRC': 'Reclutamiento Forzado',
 'LSE': 'Secuestros',
 'MAE': 'Asesinatos/Ejecuciones Extrajudiciales',
 'MAT': 'Muertos en atentados',
 'MEF': 'Muertos en enfrentamientos',
 'TLA': 'Lesiones o Heridas',
 'TTR': 'Tortura',
 'TVS': 'Violación Sexual',
 'IDPERSON': 'ID DE LA VÍCTIMA',
 'FECHALI': 'FECHA LÍMITE INFERIOR',
 'FECHALS': 'FECHA LÍMITE SUPERIOR',
 'FECHAMED': 'FECHA MEDIA',
 'ANO': 'AÑO EN EL QUE OCURRIERON LOS HECHOS',
 'SEXO': 'SEXO DE LA VÍCTIMA',
 'ANONAC': 'ANO DE NACIMIENTO',
 'EDAD': 'EDAD DE LA VÍCTIMA',
 'EDADREC': 'GRUPOS DE EDAD',
 'IDNIVEDU': 'NIVEL EDUCATIVO DE LA VÍCTIMA',
 'CONDALFA': 'CONDICIÓN DE ALFABETISMO',
 'ESTACIV': 'ESTADO CIVIL DE LA VÍCTIMA',
 'RELIGION': 'RELIGION DE LA VÍCTIMA',
 'IDIOMA': 'IDIOMA MATERNO DE LA VÍCTIMA',
 'OCUPACI0': 'OCUPACIÓN DE LA VÍCTIMA',
 'ORGANIZA': 'TIPO 

In [7]:
# Del mismo modo, extaemos las etiquetas de valor
var_values=meta.variable_value_labels
var_values

{'IDTIPOAC': {'LDS': 'Desaparición - continúa desaparecido',
  'LDT': 'Detenciones',
  'LRC': 'Reclutamiento Forzado',
  'LSE': 'Secuestros',
  'MAE': 'Asesinatos/Ejecuciones Extrajudiciales',
  'MAT': 'Muertos en atentados',
  'MEF': 'Muertos en enfrentamientos',
  'TLA': 'Lesiones o Heridas',
  'TTR': 'Tortura',
  'TVS': 'Violación Sexual'},
 'LDS_LDT': {0.0: 'No', 1.0: 'Sí'},
 'LDT': {0.0: 'No', 1.0: 'Sí'},
 'LRC': {0.0: 'No', 1.0: 'Sí'},
 'LSE': {0.0: 'No', 1.0: 'Sí'},
 'MAE': {0.0: 'No', 1.0: 'Sí'},
 'MAT': {0.0: 'No', 1.0: 'Sí'},
 'MEF': {0.0: 'No', 1.0: 'Sí'},
 'TLA': {0.0: 'No', 1.0: 'Sí'},
 'TTR': {0.0: 'No', 1.0: 'Sí'},
 'TVS': {0.0: 'No', 1.0: 'Sí'},
 'ANO': {1980.0: '80',
  1981.0: '81',
  1982.0: '82',
  1983.0: '83',
  1984.0: '84',
  1985.0: '85',
  1986.0: '86',
  1987.0: '87',
  1988.0: '88',
  1989.0: '89',
  1990.0: '90',
  1991.0: '91',
  1992.0: '92',
  1993.0: '93',
  1994.0: '94',
  1995.0: '95',
  1996.0: '96',
  1997.0: '97',
  1998.0: '98',
  1999.0: '99',
  2

## 3. Filtrar la base de datos 
Filtramos la base de datos depentiendo del tipo de acto de violencia indicado.

In [8]:
#Desaparición
cvr[cvr.IDTIPOAC=="LDS"]

IDEVENTO   IDACTO   IDACTOR IDTIPOAC  LDS_LDT  LDT  LRC  LSE  MAE  MAT  \
0      1000001  14867.0   54406.0      LDS      1.0  0.0  0.0  0.0  0.0  0.0   
1      1000001  14867.0   54407.0      LDS      1.0  0.0  0.0  0.0  0.0  0.0   
5      1000003  14837.0   54227.0      LDS      1.0  0.0  0.0  0.0  0.0  0.0   
6      1000004  14996.0   54905.0      LDS      0.0  0.0  0.0  0.0  0.0  0.0   
25     1000005  20885.0   75197.0      LDS      1.0  0.0  0.0  0.0  0.0  0.0   
...        ...      ...       ...      ...      ...  ...  ...  ...  ...  ...   
35900  1015449  34979.0  116558.0      LDS      1.0  0.0  0.0  0.0  0.0  0.0   
35903  1015451  34968.0  116536.0      LDS      1.0  0.0  0.0  0.0  0.0  0.0   
35908  1015455  35154.0  117254.0      LDS      1.0  0.0  0.0  0.0  0.0  0.0   
35940  1015518  36997.0  123515.0      LDS      1.0  0.0  0.0  0.0  0.0  0.0   
35996  1015702  37013.0  123579.0      LDS      1.0  0.0  0.0  0.0  0.0  0.0   

       ...  T08  T09  T10  T11 T12 T13 T14 T15  T16  T90  
0      ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
1      ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
5      ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
6      ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
25     ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
...    ...  ...  ...  ...  ...  ..  ..  ..  ..  ...  ...  
35900  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
35903  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
35908  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
35940  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
35996  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  

[3900 rows x 118 columns]

In [9]:
#Secuestos
cvr[cvr.IDTIPOAC=="LSE"]

IDEVENTO   IDACTO   IDACTOR IDTIPOAC  LDS_LDT  LDT  LRC  LSE  MAE  MAT  \
7      1000005  14428.0   52785.0      LSE      0.0  0.0  0.0  1.0  0.0  0.0   
8      1000005  14428.0   66646.0      LSE      0.0  0.0  0.0  1.0  0.0  0.0   
9      1000005  14428.0   66647.0      LSE      0.0  0.0  0.0  1.0  0.0  0.0   
10     1000005  14428.0   66648.0      LSE      0.0  0.0  0.0  1.0  0.0  0.0   
11     1000005  14428.0   66649.0      LSE      0.0  0.0  0.0  1.0  0.0  0.0   
...        ...      ...       ...      ...      ...  ...  ...  ...  ...  ...   
36003  1015706  37029.0  123646.0      LSE      0.0  0.0  0.0  1.0  0.0  0.0   
36004  1015706  37029.0  123647.0      LSE      0.0  0.0  0.0  1.0  0.0  0.0   
36005  1015706  37029.0  123648.0      LSE      0.0  0.0  0.0  1.0  0.0  0.0   
36006  1015706  37029.0  123649.0      LSE      0.0  0.0  0.0  1.0  0.0  0.0   
36013  1016674  35060.0  116898.0      LSE      0.0  0.0  0.0  1.0  0.0  0.0   

       ...  T08  T09  T10  T11 T12 T13 T14 T15  T16  T90  
7      ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
8      ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
9      ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
10     ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
11     ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
...    ...  ...  ...  ...  ...  ..  ..  ..  ..  ...  ...  
36003  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
36004  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
36005  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
36006  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
36013  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  

[3611 rows x 118 columns]

In [10]:
#Reclutamiento forzado
cvr[cvr.IDTIPOAC=="LRC"]

IDEVENTO   IDACTO   IDACTOR IDTIPOAC  LDS_LDT  LDT  LRC  LSE  MAE  MAT  \
93     1000040  14784.0   54005.0      LRC      0.0  0.0  1.0  0.0  0.0  0.0   
100    1000043  36730.0  122275.0      LRC      0.0  0.0  1.0  0.0  0.0  0.0   
189    1000073  15660.0   57503.0      LRC      0.0  0.0  1.0  0.0  0.0  0.0   
217    1000082  17121.0   62442.0      LRC      0.0  0.0  1.0  0.0  0.0  0.0   
272    1000109  31077.0  103067.0      LRC      0.0  0.0  1.0  0.0  0.0  0.0   
...        ...      ...       ...      ...      ...  ...  ...  ...  ...  ...   
35657  1015121  36515.0  121517.0      LRC      0.0  0.0  1.0  0.0  0.0  0.0   
35830  1015407  34592.0  115060.0      LRC      0.0  0.0  1.0  0.0  0.0  0.0   
35853  1015430  34465.0  114641.0      LRC      0.0  0.0  1.0  0.0  0.0  0.0   
35934  1015514  36846.0  122792.0      LRC      0.0  0.0  1.0  0.0  0.0  0.0   
35935  1015514  36846.0  122793.0      LRC      0.0  0.0  1.0  0.0  0.0  0.0   

       ...  T08  T09  T10  T11 T12 T13 T14 T15  T16  T90  
93     ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
100    ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
189    ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
217    ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
272    ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
...    ...  ...  ...  ...  ...  ..  ..  ..  ..  ...  ...  
35657  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
35830  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
35853  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
35934  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
35935  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  

[569 rows x 118 columns]

In [11]:
#Muertes en atentados
cvr[cvr.IDTIPOAC=="MAT"]

IDEVENTO   IDACTO   IDACTOR IDTIPOAC  LDS_LDT  LDT  LRC  LSE  MAE  MAT  \
66     1000024  15289.0   56137.0      MAT      0.0  0.0  0.0  0.0  0.0  1.0   
464    1000203   9016.0   34135.0      MAT      0.0  0.0  0.0  0.0  0.0  1.0   
465    1000203   9016.0   34136.0      MAT      0.0  0.0  0.0  0.0  0.0  1.0   
466    1000203   9016.0   34137.0      MAT      0.0  0.0  0.0  0.0  0.0  1.0   
605    1000291  10801.0   40500.0      MAT      0.0  0.0  0.0  0.0  0.0  1.0   
...        ...      ...       ...      ...      ...  ...  ...  ...  ...  ...   
35046  1014846  35509.0  118306.0      MAT      0.0  0.0  0.0  0.0  0.0  1.0   
35047  1014846  35509.0  118307.0      MAT      0.0  0.0  0.0  0.0  0.0  1.0   
35048  1014846  35509.0  118308.0      MAT      0.0  0.0  0.0  0.0  0.0  1.0   
35049  1014846  35509.0  118309.0      MAT      0.0  0.0  0.0  0.0  0.0  1.0   
35070  1014863  34867.0  116234.0      MAT      0.0  0.0  0.0  0.0  0.0  1.0   

       ...  T08  T09  T10  T11 T12 T13 T14 T15  T16  T90  
66     ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
464    ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
465    ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
466    ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
605    ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
...    ...  ...  ...  ...  ...  ..  ..  ..  ..  ...  ...  
35046  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
35047  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
35048  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
35049  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  
35070  ...  NaN  NaN  NaN  NaN NaN NaN NaN NaN  NaN  NaN  

[196 rows x 118 columns]

## 3. Borrar los duplicados
Borramos los duplicados de la columna IDACTO (ID del evento). Esto a modo de trabajar con la información de personas que sufrieron actos de violencia, en lugar de la cantidad de actos registrados.

In [12]:
#nos quedamos con la primera observacion de cada persona
cvr2 = cvr.drop_duplicates(subset=["IDACTO"], keep='first')
cvr2

IDEVENTO   IDACTO  IDACTOR IDTIPOAC  LDS_LDT  LDT  LRC  LSE  MAE  MAT  \
0      1000001  14867.0  54406.0      LDS      1.0  0.0  0.0  0.0  0.0  0.0   
2      1000001  20535.0  74109.0      LDT      0.0  1.0  0.0  0.0  0.0  0.0   
4      1000003  14834.0  54220.0      LDT      0.0  1.0  0.0  0.0  0.0  0.0   
5      1000003  14837.0  54227.0      LDS      1.0  0.0  0.0  0.0  0.0  0.0   
6      1000004  14996.0  54905.0      LDS      0.0  0.0  0.0  0.0  0.0  0.0   
...        ...      ...      ...      ...      ...  ...  ...  ...  ...  ...   
36014  1200682  13183.0  48743.0      MAE      0.0  0.0  0.0  0.0  1.0  0.0   
36015  1991433    630.0   2473.0      LDT      0.0  1.0  0.0  0.0  0.0  0.0   
36016  1991433    632.0   2478.0      TTR      0.0  0.0  0.0  0.0  0.0  0.0   
36017  1991433    640.0   2509.0      MAE      1.0  0.0  0.0  0.0  1.0  0.0   
36018  1991433    644.0   2528.0      LDT      0.0  1.0  0.0  0.0  0.0  0.0   

       ...  T08  T09  T10  T11  T12  T13  T14  T15  T16  T90  
0      ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
2      ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
4      ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
5      ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
6      ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
...    ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  
36014  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
36015  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
36016  ...  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  
36017  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
36018  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  

[20534 rows x 118 columns]

Se puede ver que ahora el número de observaciones se ha reducido

## 4. Crear distintas base de datos
### 4.1. Total de eventos de violencia según su tipo a nivel departamento

In [13]:
# Primero, buscamos qué tipo de actos se registran en la base de datos
actos=var_values['IDTIPOAC']
actos

{'LDS': 'Desaparición - continúa desaparecido',
 'LDT': 'Detenciones',
 'LRC': 'Reclutamiento Forzado',
 'LSE': 'Secuestros',
 'MAE': 'Asesinatos/Ejecuciones Extrajudiciales',
 'MAT': 'Muertos en atentados',
 'MEF': 'Muertos en enfrentamientos',
 'TLA': 'Lesiones o Heridas',
 'TTR': 'Tortura',
 'TVS': 'Violación Sexual'}

In [14]:
# Luego, buscamos los departamentos registrados
# notar que la variable DEPNA0 registra el lugar de nacimineto de la víctima
var_values['DEPNA0']

{1.0: 'AYACUCHO',
 2.0: 'APURÍMAC',
 3.0: 'HUANCAVELICA',
 4.0: 'CUSCO',
 5.0: 'HUÁNUCO',
 6.0: 'UCAYALI',
 7.0: 'SAN MARTÍN',
 8.0: 'PUNO',
 9.0: 'JUNÍN',
 10.0: 'LIMA - CALLAO',
 90.0: 'OTROS'}

In [15]:
#La base de datos ya contiene las dummys necesarias, así que solo agregamos las observaciones
dep=cvr2.groupby(['DEPNA0']).agg(LDS=('LDS_LDT', np.sum), LDT = ('LDT', np.sum), LRC = ('LRC', np.sum), LSE = ('LSE', np.sum), MAE = ('MAE', np.sum), MAT = ('MAT', np.sum), MEF = ('MEF', np.sum), TLA = ('TLA', np.sum), TTR = ('TTR', np.sum), TVS = ('TVS', np.sum))
dep

LDS     LDT   LRC    LSE     MAE   MAT   MEF    TLA    TTR    TVS
DEPNA0                                                                      
1.0     1159.0  1205.0  91.0  619.0  1895.0  15.0  76.0  192.0  996.0  122.0
2.0      146.0   316.0  31.0  131.0   274.0   2.0   7.0   24.0  378.0   32.0
3.0      176.0   246.0  22.0  103.0   356.0   7.0   8.0   40.0  235.0   33.0
4.0       29.0    84.0   3.0   22.0    71.0   0.0   4.0   11.0   80.0    7.0
5.0      520.0   490.0  57.0  345.0   802.0  14.0  41.0   49.0  487.0   35.0
6.0       48.0    54.0   2.0   25.0    33.0   3.0   4.0    4.0   37.0    2.0
7.0      124.0   120.0  12.0   67.0   196.0   8.0  13.0   23.0  118.0    6.0
8.0       36.0   102.0   5.0   41.0   132.0   6.0   4.0   19.0  135.0    2.0
9.0      241.0   251.0  30.0  211.0   474.0  10.0  16.0   39.0  216.0   22.0
10.0      79.0   234.0   6.0   38.0   111.0  10.0   4.0   28.0  165.0    7.0
90.0     154.0   594.0  14.0  114.0   359.0  15.0  22.0   70.0  539.0   30.0

In [16]:
#Editamos la DF anterior para poder ver los nombres de los departamentos
dep2=dep.rename(index={1.0: 'AYACUCHO',
 2.0: 'APURÍMAC',
 3.0: 'HUANCAVELICA',
 4.0: 'CUSCO',
 5.0: 'HUÁNUCO',
 6.0: 'UCAYALI',
 7.0: 'SAN MARTÍN',
 8.0: 'PUNO',
 9.0: 'JUNÍN',
 10.0: 'LIMA - CALLAO',
 90.0: 'OTROS'})
dep2

LDS     LDT   LRC    LSE     MAE   MAT   MEF    TLA    TTR  \
DEPNA0                                                                         
AYACUCHO       1159.0  1205.0  91.0  619.0  1895.0  15.0  76.0  192.0  996.0   
APURÍMAC        146.0   316.0  31.0  131.0   274.0   2.0   7.0   24.0  378.0   
HUANCAVELICA    176.0   246.0  22.0  103.0   356.0   7.0   8.0   40.0  235.0   
CUSCO            29.0    84.0   3.0   22.0    71.0   0.0   4.0   11.0   80.0   
HUÁNUCO         520.0   490.0  57.0  345.0   802.0  14.0  41.0   49.0  487.0   
UCAYALI          48.0    54.0   2.0   25.0    33.0   3.0   4.0    4.0   37.0   
SAN MARTÍN      124.0   120.0  12.0   67.0   196.0   8.0  13.0   23.0  118.0   
PUNO             36.0   102.0   5.0   41.0   132.0   6.0   4.0   19.0  135.0   
JUNÍN           241.0   251.0  30.0  211.0   474.0  10.0  16.0   39.0  216.0   
LIMA - CALLAO    79.0   234.0   6.0   38.0   111.0  10.0   4.0   28.0  165.0   
OTROS           154.0   594.0  14.0  114.0   359.0  15.0  22.0   70.0  539.0   

                 TVS  
DEPNA0                
AYACUCHO       122.0  
APURÍMAC        32.0  
HUANCAVELICA    33.0  
CUSCO            7.0  
HUÁNUCO         35.0  
UCAYALI          2.0  
SAN MARTÍN       6.0  
PUNO             2.0  
JUNÍN           22.0  
LIMA - CALLAO    7.0  
OTROS           30.0

### 4.2. Total de eventos de violencia según su tipo a nivel distrito

In [17]:
#Replicamos el proceso anterior
dist=cvr2.groupby(['UBIDIST']).agg(LDS=('LDS_LDT', np.sum), LDT = ('LDT', np.sum), LRC = ('LRC', np.sum), LSE = ('LSE', np.sum), MAE = ('MAE', np.sum), MAT = ('MAT', np.sum), MEF = ('MEF', np.sum), TLA = ('TLA', np.sum), TTR = ('TTR', np.sum), TVS = ('TVS', np.sum))
dist


LDS    LDT  LRC   LSE   MAE  MAT  MEF   TLA   TTR  TVS
UBIDIST                                                          
010201     1.0    2.0  0.0   0.0   0.0  0.0  0.0   0.0   1.0  0.0
010301     0.0    0.0  0.0   0.0   0.0  0.0  1.0   0.0   0.0  0.0
010403     0.0    0.0  0.0   0.0   1.0  0.0  0.0   0.0   0.0  0.0
010502     0.0    0.0  0.0   0.0   0.0  0.0  0.0   1.0   0.0  0.0
010509     0.0    1.0  0.0   0.0   0.0  0.0  0.0   0.0   0.0  0.0
...        ...    ...  ...   ...   ...  ...  ...   ...   ...  ...
250203     1.0    1.0  0.0   0.0   0.0  0.0  0.0   0.0   0.0  0.0
250300     1.0    1.0  0.0   1.0   1.0  0.0  0.0   0.0   1.0  0.0
250301   102.0  110.0  6.0  63.0  84.0  0.0  5.0  11.0  75.0  7.0
250302    27.0   27.0  0.0  14.0  35.0  0.0  2.0   4.0  20.0  2.0
250303     1.0    0.0  0.0   1.0   2.0  0.0  0.0   0.0   1.0  0.0

[919 rows x 10 columns]

In [18]:
#Calculamos el logaritmo natural de las variables
dist_log=np.log(dist)
dist_log

C:\Users\Melani\anaconda3\lib\site-packages\pandas\core\internals\blocks.py:402: RuntimeWarning: divide by zero encountered in log
  result = func(self.values, **kwargs)


LDS       LDT       LRC       LSE       MAE  MAT       MEF  \
UBIDIST                                                                    
010201   0.000000  0.693147      -inf      -inf      -inf -inf      -inf   
010301       -inf      -inf      -inf      -inf      -inf -inf  0.000000   
010403       -inf      -inf      -inf      -inf  0.000000 -inf      -inf   
010502       -inf      -inf      -inf      -inf      -inf -inf      -inf   
010509       -inf  0.000000      -inf      -inf      -inf -inf      -inf   
...           ...       ...       ...       ...       ...  ...       ...   
250203   0.000000  0.000000      -inf      -inf      -inf -inf      -inf   
250300   0.000000  0.000000      -inf  0.000000  0.000000 -inf      -inf   
250301   4.624973  4.700480  1.791759  4.143135  4.430817 -inf  1.609438   
250302   3.295837  3.295837      -inf  2.639057  3.555348 -inf  0.693147   
250303   0.000000      -inf      -inf  0.000000  0.693147 -inf      -inf   

              TLA       TTR       TVS  
UBIDIST                                
010201       -inf  0.000000      -inf  
010301       -inf      -inf      -inf  
010403       -inf      -inf      -inf  
010502   0.000000      -inf      -inf  
010509       -inf      -inf      -inf  
...           ...       ...       ...  
250203       -inf      -inf      -inf  
250300       -inf  0.000000      -inf  
250301   2.397895  4.317488  1.945910  
250302   1.386294  2.995732  0.693147  
250303       -inf  0.000000      -inf  

[919 rows x 10 columns]

### 4.3. Total de eventos de violencia según su tipo a nivel departamento y periodo

In [19]:
var_values['PERIODO']

{1.0: '80 - 82',
 2.0: '83 - 85',
 3.0: '86 - 88',
 4.0: '89 - 92',
 5.0: '93 - 00'}

In [20]:
dep_per=cvr2.groupby(['DEPNA0','PERIODO']).agg(LDS=('LDS_LDT', np.sum), LDT = ('LDT', np.sum), LRC = ('LRC', np.sum), LSE = ('LSE', np.sum), MAE = ('MAE', np.sum), MAT = ('MAT', np.sum), MEF = ('MEF', np.sum), TLA = ('TLA', np.sum), TTR = ('TTR', np.sum), TVS = ('TVS', np.sum))
dep_per

LDS    LDT   LRC    LSE    MAE   MAT   MEF   TLA    TTR  \
DEPNA0 PERIODO                                                              
1.0    1.0       49.0   46.0   6.0   51.0  112.0   1.0   4.0   7.0   54.0   
       2.0      748.0  715.0  44.0  336.0  978.0   2.0  27.0  88.0  497.0   
       3.0      150.0  176.0  12.0   96.0  296.0   6.0  11.0  29.0  173.0   
       4.0      185.0  196.0  26.0  119.0  433.0   6.0  22.0  49.0  211.0   
       5.0       27.0   72.0   3.0   17.0   76.0   0.0  12.0  19.0   61.0   
2.0    1.0        6.0   24.0   1.0    8.0   12.0   0.0   0.0   3.0   24.0   
       2.0       29.0   50.0   3.0   26.0   37.0   0.0   0.0   2.0   56.0   
       3.0       52.0  103.0  13.0   44.0   92.0   0.0   2.0  10.0  130.0   
       4.0       45.0   95.0  11.0   44.0  107.0   2.0   5.0   7.0  128.0   
       5.0       14.0   44.0   3.0    9.0   26.0   0.0   0.0   2.0   40.0   
3.0    1.0        3.0    4.0   2.0    4.0   12.0   0.0   0.0   1.0    8.0   
       2.0       50.0   77.0   8.0   32.0  105.0   0.0   2.0  13.0   79.0   
       3.0       19.0   27.0   5.0   19.0   52.0   1.0   1.0   6.0   28.0   
       4.0       97.0   98.0   4.0   41.0  155.0   4.0   4.0  17.0   88.0   
       5.0        7.0   40.0   3.0    7.0   32.0   2.0   1.0   3.0   32.0   
4.0    1.0        1.0    6.0   0.0    2.0    1.0   0.0   0.0   0.0    7.0   
       2.0       11.0   15.0   1.0    9.0   14.0   0.0   1.0   2.0   13.0   
       3.0        4.0    9.0   0.0    6.0   18.0   0.0   0.0   2.0   14.0   
       4.0       11.0   26.0   2.0    3.0   30.0   0.0   3.0   7.0   26.0   
       5.0        2.0   28.0   0.0    2.0    8.0   0.0   0.0   0.0   20.0   
5.0    1.0        4.0    3.0   1.0    6.0   15.0   0.0   0.0   2.0    4.0   
       2.0       49.0   50.0  11.0   27.0   95.0   0.0   3.0   6.0   43.0   
       3.0       79.0   45.0  18.0   69.0  144.0   5.0   9.0   3.0   69.0   
       4.0      235.0  213.0  17.0  126.0  291.0   7.0  10.0  19.0  184.0   
       5.0      153.0  179.0  10.0  117.0  257.0   2.0  19.0  19.0  187.0   
6.0    2.0        0.0    1.0   0.0    0.0    1.0   0.0   0.0   0.0    1.0   
       3.0        3.0    4.0   1.0    2.0    4.0   1.0   0.0   0.0    4.0   
       4.0       38.0   37.0   1.0   16.0   20.0   2.0   3.0   3.0   22.0   
       5.0        7.0   12.0   0.0    7.0    8.0   0.0   1.0   1.0   10.0   
7.0    1.0        0.0    1.0   0.0    0.0    2.0   0.0   0.0   1.0    1.0   
       2.0        2.0    2.0   3.0    1.0    6.0   2.0   0.0   0.0    3.0   
       3.0       24.0   16.0   2.0   15.0   23.0   2.0   1.0   1.0   11.0   
       4.0       83.0   75.0   6.0   41.0  104.0   3.0   7.0  14.0   69.0   
       5.0       15.0   26.0   1.0   10.0   61.0   1.0   5.0   7.0   34.0   
8.0    1.0        0.0    1.0   0.0    0.0    0.0   0.0   0.0   1.0    1.0   
       2.0        1.0    2.0   0.0    0.0    3.0   1.0   0.0   0.0    4.0   
       3.0        4.0   12.0   1.0    6.0   12.0   1.0   0.0   5.0   17.0   
       4.0       28.0   27.0   4.0   29.0   81.0   4.0   4.0   9.0   55.0   
       5.0        3.0   60.0   0.0    6.0   36.0   0.0   0.0   4.0   58.0   
9.0    1.0        1.0    0.0   0.0    0.0    2.0   1.0   0.0   0.0    0.0   
       2.0        8.0   16.0   3.0    7.0   14.0   1.0   0.0   5.0   11.0   
       3.0       21.0   17.0   7.0   28.0   44.0   0.0   3.0   4.0   17.0   
       4.0      186.0  167.0  19.0  154.0  349.0   6.0   9.0  25.0  145.0   
       5.0       25.0   51.0   1.0   22.0   65.0   2.0   4.0   5.0   43.0   
10.0   1.0        0.0    4.0   0.0    1.0    1.0   0.0   0.0   0.0    3.0   
       2.0       14.0   23.0   2.0    6.0   14.0   0.0   1.0   3.0   14.0   
       3.0        5.0   13.0   1.0    6.0   21.0   1.0   1.0   6.0   16.0   
       4.0       45.0   92.0   3.0   19.0   61.0   6.0   1.0  19.0   50.0   
       5.0       15.0  102.0   0.0    6.0   14.0   3.0   1.0   0.0   82.0   
90.0   1.0        2.0   11.0   1.0    3.0    6.0   0.0   0.0   1.0    8.0   
       2.0    